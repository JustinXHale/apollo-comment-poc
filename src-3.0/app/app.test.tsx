import * as React from 'react';
import App from '@app/index';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';

// Mock localStorage
const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
};
Object.defineProperty(window, 'localStorage', {
  value: localStorageMock
});

describe('App tests', () => {
  beforeEach(() => {
    // Set up authentication for all tests
    localStorageMock.getItem.mockImplementation((key) => {
      if (key === 'rhoai-auth') return 'authenticated';
      return null;
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render default App component', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a nav-toggle button', () => {
    render(<App />);

    expect(screen.getByRole('button', { name: 'Global navigation' })).toBeVisible();
  });

  // I'm fairly sure that this test not going to work properly no matter what we do since JSDOM doesn't actually
  // draw anything. We could potentially make something work, likely using a different test environment, but
  // using Cypress for this kind of test would be more efficient.
  it.skip('should hide the sidebar on smaller viewports', async () => {
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 600 });

    render(<App />);

    await act(async () => {
      window.dispatchEvent(new Event('resize'));
    });

    expect(screen.queryByRole('button', { name: 'Generative AI studio' })).not.toBeInTheDocument();
  });

  it('should expand the sidebar on larger viewports', async () => {
    render(<App />);

    await act(async () => {
      window.dispatchEvent(new Event('resize'));
    });

    expect(screen.getByRole('button', { name: 'Generative AI studio' })).toBeVisible();
  });

  it('should hide the sidebar when clicking the nav-toggle button', async () => {
    const user = userEvent.setup();

    render(<App />);

    await act(async () => {
      window.dispatchEvent(new Event('resize'));
    });
    const button = screen.getByRole('button', { name: 'Global navigation' });

    expect(screen.getByRole('button', { name: 'Generative AI studio' })).toBeVisible();

    await user.click(button);

    expect(screen.queryByRole('button', { name: 'Generative AI studio' })).not.toBeInTheDocument();
  });
});
