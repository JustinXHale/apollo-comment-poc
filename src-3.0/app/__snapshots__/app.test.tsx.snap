// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<DocumentFragment>
  <div
    style="height: 100vh; display: flex; flex-direction: column;"
  >
    <div
      style="background-color: rgb(255, 111, 0); color: white; text-align: center; padding: 0.25rem; font-size: 0.75rem; font-weight: 600; letter-spacing: 0.5px; flex-shrink: 0; z-index: 1000; display: flex; align-items: center; justify-content: center; gap: 0.5rem;"
    >
      UXD PROTOTYPE
      <div
        style="display: contents;"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          style="font-size: 0.75rem; cursor: pointer; opacity: 0.8;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
          />
        </svg>
      </div>
    </div>
    <div
      class="pf-v6-c-page"
      style="flex: 1; min-height: 0;"
    >
      <div
        class="pf-v6-c-skip-to-content"
      >
        <a
          class="pf-v6-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-2"
          data-ouia-component-type="PF6/Button"
          data-ouia-safe="true"
          href="#primary-app-container"
        >
          <span
            class="pf-v6-c-button__text"
          >
            Skip to Content
          </span>
        </a>
      </div>
      <header
        class="pf-v6-c-masthead pf-m-display-inline-on-md"
      >
        <div
          class="pf-v6-c-masthead__main"
        >
          <span
            class="pf-v6-c-masthead__toggle"
          >
            <button
              aria-label="Global navigation"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  />
                </svg>
              </span>
            </button>
          </span>
          <div
            class="pf-v6-c-masthead__brand"
            style="display: flex; align-items: center;"
          >
            <img
              alt="Product Logo Light"
              src="test-file-stub"
              style="height: 36px; width: auto; max-width: 200px;"
            />
          </div>
        </div>
        <div
          class="pf-v6-c-masthead__content"
        >
          <div
            style="position: absolute; right: 1rem; top: 50%; transform: translateY(-50%); display: flex; align-items: center; gap: 0.25rem; z-index: 10;"
          >
            <button
              aria-expanded="false"
              aria-label="Fidelity switcher"
              class="pf-v6-c-menu-toggle pf-m-plain"
              data-ouia-component-id="OUIA-Generated-MenuToggle-plain-1"
              data-ouia-component-type="PF6/MenuToggle"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-menu-toggle__text"
              >
                <span
                  style="display: flex; align-items: center; gap: 0.5rem;"
                >
                  High fidelity
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <button
              aria-label="Alerts"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-2"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 896 1024"
                  width="1em"
                >
                  <path
                    d="M448,0 C465.333333,0 480.333333,6.33333333 493,19 C505.666667,31.6666667 512,46.6666667 512,64 L512,106 L514.23,106.45 C587.89,121.39 648.48,157.24 696,214 C744,271.333333 768,338.666667 768,416 C768,500 780,568.666667 804,622 C818.666667,652.666667 841.333333,684 872,716 C873.773676,718.829136 875.780658,721.505113 878,724 C890,737.333333 896,752.333333 896,769 C896,785.666667 890,800.333333 878,813 C866,825.666667 850.666667,832 832,832 L63.3,832 C44.9533333,831.84 29.8533333,825.506667 18,813 C6,800.333333 0,785.666667 0,769 C0,752.333333 6,737.333333 18,724 L24,716 L25.06,714.9 C55.1933333,683.28 77.5066667,652.313333 92,622 C116,568.666667 128,500 128,416 C128,338.666667 152,271.333333 200,214 C248,156.666667 309.333333,120.666667 384,106 L384,63.31 C384.166667,46.27 390.5,31.5 403,19 C415.666667,6.33333333 430.666667,0 448,0 Z M576,896 L576,897.08 C575.74,932.6 563.073333,962.573333 538,987 C512.666667,1011.66667 482.666667,1024 448,1024 C413.333333,1024 383.333333,1011.66667 358,987 C332.666667,962.333333 320,932 320,896 L576,896 Z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Information"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-3"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Switch to dark mode"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-4"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-expanded="false"
              aria-label="User menu"
              class="pf-v6-c-menu-toggle pf-m-plain"
              data-ouia-component-id="OUIA-Generated-MenuToggle-plain-2"
              data-ouia-component-type="PF6/MenuToggle"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-menu-toggle__text"
              >
                <div
                  style="display: flex; align-items: center; gap: 0.5rem; padding: 0.25rem 0.5rem;"
                >
                  <div
                    style="width: 24px; height: 24px; background-color: rgb(0, 102, 204); color: white; border-radius: 50%; display: flex; align-items: center; justify-content: center; font-size: 12px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                      />
                    </svg>
                  </div>
                  <span
                    style="font-size: 14px; font-weight: 500;"
                  >
                    AI Engineer
                  </span>
                </div>
              </span>
            </button>
          </div>
        </div>
      </header>
      <div
        aria-hidden="false"
        class="pf-v6-c-page__sidebar pf-m-expanded"
        id="page-sidebar"
      >
        <div
          class="pf-v6-c-page__sidebar-body"
        >
          <nav
            aria-label="Global"
            class="pf-v6-c-nav"
            data-ouia-component-id="OUIA-Generated-Nav-1"
            data-ouia-component-type="PF6/Nav"
            data-ouia-safe="true"
            id="nav-primary-simple"
          >
            <ul
              class="pf-v6-c-nav__list"
              id="nav-list-simple"
              role="list"
            >
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-1"
                data-ouia-component-type="PF6/NavItem"
                data-ouia-safe="true"
              >
                <div
                  aria-current="page"
                  class="pf-v6-c-nav__link pf-m-current"
                  style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                >
                  <span>
                    Home
                  </span>
                </div>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-2"
                data-ouia-component-type="PF6/NavItem"
                data-ouia-safe="true"
              >
                <div
                  class="pf-v6-c-nav__link"
                  style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                >
                  <span>
                    Projects
                  </span>
                </div>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavExpandable-1"
                data-ouia-component-type="PF6/NavExpandable"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  class="pf-v6-c-nav__link"
                  id="AI hub-2"
                >
                  AI hub
                  <span
                    class="pf-v6-c-nav__toggle"
                  >
                    <span
                      class="pf-v6-c-nav__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v6-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <section
                  aria-labelledby="AI hub-2"
                  class="pf-v6-c-nav__subnav"
                  hidden=""
                  inert=""
                >
                  <ul
                    class="pf-v6-c-nav__list"
                    role="list"
                  >
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-3"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <a
                        class="pf-v6-c-nav__link"
                        data-discover="true"
                        href="/ai-hub/catalog"
                        style="display: flex; align-items: center; justify-content: space-between;"
                      >
                        <span>
                          Catalog
                        </span>
                      </a>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-4"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <a
                        class="pf-v6-c-nav__link"
                        data-discover="true"
                        href="/ai-hub/registry"
                        style="display: flex; align-items: center; justify-content: space-between;"
                      >
                        <span>
                          Registry
                        </span>
                      </a>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-5"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <a
                        class="pf-v6-c-nav__link"
                        data-discover="true"
                        href="/ai-hub/deployments"
                        style="display: flex; align-items: center; justify-content: space-between;"
                      >
                        <span>
                          Deployments
                        </span>
                      </a>
                    </li>
                  </ul>
                </section>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavExpandable-2"
                data-ouia-component-type="PF6/NavExpandable"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  class="pf-v6-c-nav__link"
                  id="Generative AI studio-4"
                >
                  Generative AI studio
                  <span
                    class="pf-v6-c-nav__toggle"
                  >
                    <span
                      class="pf-v6-c-nav__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v6-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <section
                  aria-labelledby="Generative AI studio-4"
                  class="pf-v6-c-nav__subnav"
                  hidden=""
                  inert=""
                >
                  <ul
                    class="pf-v6-c-nav__list"
                    role="list"
                  >
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-6"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <a
                        class="pf-v6-c-nav__link"
                        data-discover="true"
                        href="/ai-assets/available"
                        style="display: flex; align-items: center; justify-content: space-between;"
                      >
                        <span>
                          AI asset endpoints
                        </span>
                        <span
                          class="pf-v6-c-label pf-m-orange pf-m-outline pf-m-compact"
                          style="font-size: 10px;"
                        >
                          <span
                            class="pf-v6-c-label__content"
                          >
                            <span
                              class="pf-v6-c-label__text"
                            >
                              Tech Preview
                            </span>
                          </span>
                        </span>
                      </a>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-7"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <a
                        class="pf-v6-c-nav__link"
                        data-discover="true"
                        href="/gen-ai-studio/ai-playground"
                        style="display: flex; align-items: center; justify-content: space-between;"
                      >
                        <span>
                          Playground
                        </span>
                        <span
                          class="pf-v6-c-label pf-m-orange pf-m-outline pf-m-compact"
                          style="font-size: 10px;"
                        >
                          <span
                            class="pf-v6-c-label__content"
                          >
                            <span
                              class="pf-v6-c-label__text"
                            >
                              Tech Preview
                            </span>
                          </span>
                        </span>
                      </a>
                    </li>
                  </ul>
                </section>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavExpandable-3"
                data-ouia-component-type="PF6/NavExpandable"
                data-ouia-safe="true"
                style="color: rgb(106, 110, 115); opacity: 0.5; cursor: not-allowed;"
              >
                <button
                  aria-expanded="false"
                  class="pf-v6-c-nav__link"
                  id="Develop & train-5"
                >
                  Develop & train
                  <span
                    class="pf-v6-c-nav__toggle"
                  >
                    <span
                      class="pf-v6-c-nav__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v6-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <section
                  aria-labelledby="Develop & train-5"
                  class="pf-v6-c-nav__subnav"
                  hidden=""
                  inert=""
                >
                  <ul
                    class="pf-v6-c-nav__list"
                    role="list"
                  >
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-8"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-v6-c-nav__link"
                        style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                      >
                        <span>
                          Pipelines
                        </span>
                      </div>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-9"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-v6-c-nav__link"
                        style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                      >
                        <span>
                          Evaluations
                        </span>
                      </div>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-10"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-v6-c-nav__link"
                        style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                      >
                        <span>
                          Experiments
                        </span>
                      </div>
                    </li>
                  </ul>
                </section>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavExpandable-4"
                data-ouia-component-type="PF6/NavExpandable"
                data-ouia-safe="true"
                style="color: rgb(106, 110, 115); opacity: 0.5; cursor: not-allowed;"
              >
                <button
                  aria-expanded="false"
                  class="pf-v6-c-nav__link"
                  id="Observe & monitor-6"
                >
                  Observe & monitor
                  <span
                    class="pf-v6-c-nav__toggle"
                  >
                    <span
                      class="pf-v6-c-nav__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v6-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <section
                  aria-labelledby="Observe & monitor-6"
                  class="pf-v6-c-nav__subnav"
                  hidden=""
                  inert=""
                >
                  <ul
                    class="pf-v6-c-nav__list"
                    role="list"
                  >
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-11"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-v6-c-nav__link"
                        style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                      >
                        <span>
                          Workload metrics
                        </span>
                      </div>
                    </li>
                    <li
                      class="pf-v6-c-nav__item"
                      data-ouia-component-id="OUIA-Generated-NavItem-12"
                      data-ouia-component-type="PF6/NavItem"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-v6-c-nav__link"
                        style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                      >
                        <span>
                          Training jobs
                        </span>
                      </div>
                    </li>
                  </ul>
                </section>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-13"
                data-ouia-component-type="PF6/NavItem"
                data-ouia-safe="true"
              >
                <div
                  class="pf-v6-c-nav__link"
                  style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                >
                  <span>
                    Learning resources
                  </span>
                </div>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-14"
                data-ouia-component-type="PF6/NavItem"
                data-ouia-safe="true"
              >
                <div
                  class="pf-v6-c-nav__link"
                  style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                >
                  <span>
                    Applications
                  </span>
                </div>
              </li>
              <li
                class="pf-v6-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-15"
                data-ouia-component-type="PF6/NavItem"
                data-ouia-safe="true"
              >
                <div
                  class="pf-v6-c-nav__link"
                  style="display: flex; align-items: center; justify-content: space-between; padding: 0.5rem 1rem; color: rgb(106, 110, 115); cursor: not-allowed; opacity: 0.5;"
                >
                  <span>
                    Settings
                  </span>
                </div>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <div
        class="pf-v6-c-page__main-container"
      >
        <main
          class="pf-v6-c-page__main"
          id="primary-app-container"
          tabindex="-1"
        >
          <section
            class="pf-v6-c-page__main-section pf-m-fill"
          >
            <div
              class="pf-v6-c-page__main-body"
            >
              <div
                class="pf-v6-l-flex pf-m-align-items-center pf-m-justify-content-center"
                style="height: 60vh; text-align: center;"
              >
                <div
                  class=""
                >
                  <div>
                    <p
                      style="font-size: 1.125rem; color: rgb(106, 110, 115); max-width: 600px; line-height: 1.6; margin-bottom: 1rem;"
                    >
                      The focus of this prototype is 
                      <button
                        class="pf-v6-c-button pf-m-link pf-m-inline"
                        data-ouia-component-id="OUIA-Generated-Button-link-1"
                        data-ouia-component-type="PF6/Button"
                        data-ouia-safe="true"
                        style="font-weight: bold; padding: 0px;"
                        type="button"
                      >
                        <span
                          class="pf-v6-c-button__text"
                        >
                          AI Hub
                        </span>
                      </button>
                       and 
                      <button
                        class="pf-v6-c-button pf-m-link pf-m-inline"
                        data-ouia-component-id="OUIA-Generated-Button-link-2"
                        data-ouia-component-type="PF6/Button"
                        data-ouia-safe="true"
                        style="font-weight: bold; padding: 0px;"
                        type="button"
                      >
                        <span
                          class="pf-v6-c-button__text"
                        >
                          Generative AI Studio
                        </span>
                      </button>
                      .
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
